coldfront/core/project/views.py

```diff
@@ -1075,53 +1075,64 @@ def post(self, request, *args, **kwargs):
 
             project_user_update_form = ProjectUserUpdateForm(request.POST,
                                                              initial={'role': project_user_obj.role.name,
                                                                       'enable_notifications': project_user_obj.enable_notifications})
 
+            if project_user_obj.role.name == 'Principal Investigator':
+                enable_notifications = project_user_update_form.data.get('enable_notifications', 'off') == 'on'
+
+                # cannot disable when no manager(s) exists
+                if not managers.exists() and not enable_notifications:
+                    messages.error(request, 'PIs can disable notifications when atleast one manager exists.')
+                else:
+                    project_user_obj.enable_notifications = enable_notifications
+                    project_user_obj.save()
+                    messages.success(request, 'User details updated.')
+
+                return HttpResponseRedirect(reverse('project-user-detail', kwargs={'pk': project_obj.pk, 'project_user_pk': project_user_obj.pk}))
+
             if project_user_update_form.is_valid():
                 form_data = project_user_update_form.cleaned_data
-                project_user_obj.enable_notifications = form_data.get('enable_notifications')
+                enable_notifications = form_data.get('enable_notifications')
 
                 old_role = project_user_obj.role
                 new_role = ProjectUserRoleChoice.objects.get(name=form_data.get('role'))
-                only_manager = not project_obj.projectuser_set.filter(~Q(pk=project_user_pk),
-                                                                      role__name='Manager').exists()
-
-                if old_role.name == 'Manager' and only_manager:
-                    if new_role.name == 'User':
-                        project_pis = project_obj.projectuser_set.filter(role__name='Principal Investigator')
-
-                        if not project_pis.exists():
-                            # no pis exist, cannot demote
-                            new_role = old_role
-                            messages.error(
-                                request, 'The project must have at least one PI or manager with notifications enabled.')
-                        else:
-                            messages.warning(request, 'User {} is no longer a manager. All PIs will now receive notifications.'
-                                             .format(project_user_obj.user.username))
-                            for pi in project_pis:
-                                pi.enable_notifications = True
-                                pi.save()
 
-                    elif new_role.name == 'Principal Investigator':
-                        project_user_obj.enable_notifications = True
+                # demote the only manager to user role
+                if old_role.name == 'Manager' and new_role.name == 'User' and\
```

Comment:
As a manger, when I demoted myself to a user, I got 403 Forbidden as a result, since it redirects to the same page after the change, and users can't view it. Can you reproduce that behavior?

It might be helpful to redirect to avoid this.